### PR TITLE
added the linkToMessage property for Teams MessageActionsPayload class

### LINKED ISSUE #4062: Teams `MessageActionsPayload` class is missing `linkToMessage` property
## Version
Bot Builder .NET SDK 4.9.3

## Describe the bug
Teams `MessageActionsPayload` class is missing the `linkToMessage` property.

The Generated folder in the master branch seems to be updated 9 months ago and could be the reason why it doesn't have the linkToMessage property.

## To Reproduce
Observe the MessageActionsPayload class: https://github.com/microsoft/botbuilder-dotnet/blob/master/libraries/Microsoft.Bot.Schema/Teams/Generated/MessageActionsPayload.cs

## Expected behavior
LinkToMessage property should be present in the MessageActionsPayload class.

## Screenshots
image

[bug]

### PR DESCRIPTION
**Pull request by vman**
Fixes #4062

## Description
Added the LinkToMessage property for Teams MessageActionsPayload class. The Bot Framework REST API sends this property in the JSON payload but the SDK was missing it.

## Specific Changes
Updated the `MessageActionsPayload` class with the  missing `LinkToMessage` property

### CODE REVIEW

File: libraries/Microsoft.Bot.Schema/Teams/Generated/MessageActionsPayload.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// <auto-generated>
+﻿// <auto-generated>
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Schema/Teams/Generated/MessageActionsPayload.cs:1]: This line can be deleted, as this was initially auto-generated, but from here on out is hand-coded.

File: libraries/Microsoft.Bot.Schema/Teams/Generated/MessageActionsPayload.cs [hunk 2]
```diff
@@ -60,7 +60,8 @@ public MessageActionsPayload()
...
+        /// <param name="linkToMessage">Link to the message on which the action was invoked</param>
+        public MessageActionsPayload(string id = default(string), string replyToId = default(string), string messageType = default(string), string createdDateTime = default(string), string lastModifiedDateTime = default(string), bool? deleted = default(bool?), string subject = default(string), string summary = default(string), string importance = default(string), string locale = default(string), MessageActionsPayloadFrom from = default(MessageActionsPayloadFrom), MessageActionsPayloadBody body = default(MessageActionsPayloadBody), string attachmentLayout = default(string), IList<MessageActionsPayloadAttachment> attachments = default(IList<MessageActionsPayloadAttachment>), IList<MessageActionsPayloadMention> mentions = default(IList<MessageActionsPayloadMention>), IList<MessageActionsPayloadReaction> reactions = default(IList<MessageActionsPayloadReaction>), string linkToMessage = default(string))
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Schema/Teams/Generated/MessageActionsPayload.cs:64]: @vman Unfortunately, adding a new (even default) parameter to the end of a method breaks .Net binary compatibility, which is a requirement we have. Even adding an overload won't work here, as all the default argument will create ambiguity. 

In the near term, we need to add an overload that takes a POCO class with all the properties. Maybe we should do that here? something like:
```
public MessageActionPayload(TeamsMessageActionPayloads paylods){ };
public class TeamsMessageActionPayloads
{
  public string Id  { get; set; } = default(string);
   ...
  public string LinkToMessage { get; set; } = default(string); 
}

Comment (cleemullins) [reply libraries/Microsoft.Bot.Schema/Teams/Generated/MessageActionsPayload.cs:64]: The other (bad) option is to add the property as you've done below, but don't have a constructor parameter for it.

**Comment by vman** [comment]
Thanks for your feedback @cleemullins , I will have have a look and get back to you.

**Comment by cleemullins** [comment]
@gabog Gabo believes this work may already have landed in a seperate PR.

**Comment by gabog** [comment]
Hi @vman, thanks a lot for the submission, we addressed this for R10 as part of https://github.com/microsoft/botbuilder-dotnet/issues/4120 and it should available in our next release in august.

**Pull request closed without merging**